src/mangle.c

```diff
@@ -901,3 +901,26 @@ void mangleToBuffer(Expression *e, OutBuffer *buf)
     Mangler v(buf);
     e->accept(&v);
 }
+
+void mangleToFuncSignature(OutBuffer *buf, FuncDeclaration *fd)
```

Comment:
As I commented to, overload conflict check does not need function return type. Even if two auto return functions defined in non-root module, their conflicts can be detected without their semantic3.

```d
auto foo() { ... }
auto foo() { ... }   // compiler can detect conflict, without semantic3 running
```